include/networkit/distance/AStar.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * AStar.hpp
+ *
+ *  Created on: 09.08.2019
+ *      Author: Eugenio Angriman <angrimae@hu-berlin.de>
+ */
+
+#ifndef A_STAR_H
+#define A_STAR_H
+
+#include <networkit/distance/AStarGeneral.hpp>
+
+/**
+ * @ingroup distance
+ * AStar path-finding algoritm
+ */
+namespace NetworKit {
+class AStar final : public STSP {
+public:
+    /**
+     * Creates the AStarGeneral class for the graph @a G, the source node @a
+     * source, and the target node @a target using a template parameter as
+     * heuristic function.
+     *
+     * @param G The graph.
+     * @param distanceHeu Vector with a lower bound of the distance from every
+     * node to the target.
+     * @param source The source node.
+     * @param target The target node.
+     * @param storePred If true, the algorithm will also store the predecessors
+     * and reconstruct a shortest path from @a source and @a target.
+     */
+    AStar(const Graph &G, const std::vector<double> &distanceHeu, node source, node target,
+                 bool storePred = true)
+        : STSP(G, source, target, storePred), distanceHeu(distanceHeu),
+          astar(AStarGeneral<Heuristic>(G, Heuristic(distanceHeu), source, target, storePred)) {}
+
+    /*
+     * Executes the AStar algorithm.
+     */
+    void run() override {
+        astar.run();
+        hasRun = true;
+    }
+
+    edgeweight getDistance() const override { return astar.getDistance(); }
+
+    std::vector<node> getPath() const { return astar.getPath(); }
+
```

Comment:
`std::vector<node> getPredecessors() const` is inherited from `STSP` but is not forwarded as getDistance. Also observe that neither getDistance nor getPredecessor are virtual in STSP, so we might get unexcepted behaivor ...

Reply:
Do you have some suggestions? Also, `getDistance` is virtual in STSP, while `getPredecessor` is not.

Reply:
Should we make it virtual in STSP and mark all implementations final? Then virtual dispatch only occurs, if one uses generic STSP types?

Reply:
Just so we're not in a deadlock waiting on each other. This issue is the only thing stopping me from approving the PR. Should we just make `STSP::getPath` virtual and mark `AStar::getPath` as override (it's the only method overriding; all other implementations rely on the generic impl).

Reply:
`AStar::getPath` does not do anything different than `STSP::getPath`, so I think that the best solution here is to just delete `AStar::getPath`.

Reply:
That's true. I made `STSP::getPath`virtual and marked `AStar::getPath` as override.